### ISSUE TITLE
switch to app login for geni

### ISSUE THREAD

**Issue by cz172638** [Thimxx/pyGenealogical-Tools#71]
geni [app login](https://www.geni.com/platform/developer/help/oauth_app_login?version=1) should be another approach. each user has to register own application and get client_id and client secret. regeneration of new token can be done programatically, without going to web ui

**Comment by Thimxx** [comment]
The problem is that it will expose the application keys to the world. But I can change the login procedure.

**Comment by cz172638** [comment]
not at all, everybody, who wants to use it, have to register own client_id and client secret

**Comment by Thimxx** [comment]
This authorization flow is only available for public data, it cannot be used for restricted data. It is described [here](https://www.geni.com/platform/developer/help/oauth_intro?version=1). Saying the following:

> Use this flow if your application needs to retrieve public data or take adminstrative actions. The calls will not be made on behalf of the user, but on behalf of the app itself. Unlike public api calls without a token, this flow has a higher rate limit.

The ideal one is the "[Trusted Authorization Flow](https://www.geni.com/platform/developer/help/oauth_trusted_client)" but for that it is needed the approval of Geni and it is per application.

**Comment by cz172638** [comment]
hmm, i used it for playing with my data - not public. application for public data gets higher limits.

what i mean, i have to pass also userid and password to get access to data.

but not sure whethere something changed as i have not played with geni few months

**Comment by Thimxx** [comment]
Do you have a code example of how you did it? I tried and it was not possile without geni authorization
